### PR TITLE
Add IsUsernameBlocked client request to Helios

### PR DESCRIPTION
**Pull request by drsnyder**
The corresponding MediaWiki change is [here](https://github.com/Wikia/app/pull/8098). This adds the client functionality via `IsUsernameBlocked` in the `wikia` package.

https://wikia-inc.atlassian.net/browse/SERVICES-595

/cc @Wikia/services-team

### CODE REVIEW

File: wikia/wikia.go [hunk 1]
```diff
@@ -118,10 +126,39 @@ func (client *Client) SendPasswordResetEmail(name, newPassword string) (bool, er
...
+	data.Add(usernameKey, name)
+
+	payload := &BlockedResponse{}
+	response, err := client.callWithToken(endpoint, data, response)
+	if err != nil {
+		return false, err
+	}
+
+	if response.StatusCode == http.StatusOK {
+		return payload.Blocked, nil
```
Comment (nmonterroso) [inline wikia/wikia.go]: does this have to be json_decoded into? right now it creates a `BlockedResponse` but never writes to it, so it should always be whatever the default for a `bool` is (false?)

**Comment by nmonterroso** [comment]
are tests part of a different ticket?

**Comment by ArturKlajnerok** [comment]
Lgtm :+1:

File: wikia/wikia.go (hunk 1 above)
Comment (drsnyder) [inline wikia/wikia.go]: Oh, interesting. Good catch. This is a typo. I guess that's the problem with passing a writable interface to the function.

**Comment by drsnyder** [comment]
> are tests part of a different ticket?

Good call. Currently, none of the code in this package is tested.

Let's change that.

File: wikia/wikia.go (hunk 1 above)
Comment (drsnyder) [inline wikia/wikia.go]: Fix this specific issue in 9fed23c. I created https://wikia-inc.atlassian.net/browse/SERVICES-622 for improving the testability of this package.

**Comment by drsnyder** [comment]
In standup we agreed to merge this. It will be used with what @ArturKlajnerok is implementing to complete the password reset flow.

**Pull request merged**